cachito/web/utils.py

```diff
@@ -21,12 +19,10 @@ def deep_sort_icm(orig_item):
     if isinstance(orig_item, list):
         sorted_item = [deep_sort_icm(item) for item in orig_item]
         # If item is a list of dicts with the "purl" key, sort by the "purl" value
-        if sorted_item and isinstance(sorted_item[0], OrderedDict) and "purl" in sorted_item[0]:
+        if sorted_item and isinstance(sorted_item[0], dict) and "purl" in sorted_item[0]:
             sorted_item = sorted(sorted_item, key=lambda item: item["purl"])
     elif isinstance(orig_item, dict):
-        items = ((k, deep_sort_icm(v)) for k, v in orig_item.items())
-        # Sort items by key
-        sorted_item = OrderedDict(sorted(items, key=lambda keyval: keyval[0]))
+        sorted_item = {k: deep_sort_icm(v) for k, v in orig_item.items()}
```

Comment:
How about changing `deep_sort_icm` to sort in place, without making copies of anything?
```python
def sort_icm_in_place(icm: dict) -> None
```
Wouldn't that reduce the memory usage of this function itself to practically 0? (Not quite, Python has loads of overhead for everything, but close).